W-10673838 veeva vault clean up (#1802)
mulesoft/docs-connectors: veevavault/1.3/modules/ROOT/pages/veevavault-connector-studio.adoc

> = Use Studio to Configure Veeva Vault Connector 1.3 - Mule 4

Reviewer: Per CX template, this would be "Using...."

You can configure one of these sources to use with Veeva Vault Connector:

* HTTP Listener +
Initiates a flow each time it receives a request on the configured host and port
* Scheduler +
Initiates a flow when a time-based condition is met

For example, to configure an HTTP Listener source, follow these steps:

. In *Mule Palette*, select *HTTP > Listener*.
. Drag *Listener* to the Studio canvas.
. On the *Listener* configuration screen, optionally change the value of the *Display Name* field.
. Specify a value for the *Path* field.
. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of the HTTP Listener source in the app.
. On the *General* tab, specify connection information for the connector.
. On the *TLS* tab, optionally specify the TLS information for the connector.
. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

[[add-connector-operation]]
== Add a Connector Operation to the Flow

> When you add a connector operation to your flow, you are specifying an action for that connector to perform.

Reviewer: I wonder if we should amend the connector template to instruct writers to name some example operations specific to the connector being documented? What do you think?

Author: Yes! We do that with sources so it would be a good idea to do that with operations.

To add an operation for Veeva Vault Connector, follow these steps:

. In *Mule Palette*, select *Veeva Vault* and then select the desired operation.
. Drag the operation onto the Studio canvas, next to the input source.

[[configure-global-element]]
== Configure a Global Element for the Connector

When you configure a connector, configure a global element that all instances of that connector in the app can use. Configuring a global element requires you to provide the authentication credentials that the connector requires to access the target Veeva Vault system. Veeva Vault Connector supports Basic authentication.

To configure the global element for Veeva Vault Connector, follow these steps:

> . Select the operation in the Studio canvas.

Reviewer: Is this "the configuration operation"? Should we alter the template to specify?

Author: Should just be operation

. In the *General* tab, in *Connection*, select the <<basic-auth, Veeva Vault Config>> method to configure Basic authentication.
+
You can reference a configuration file that contains ANT-style property placeholders (recommended), or you can enter your authorization credentials in the global configuration properties. For information about the benefits of using property placeholders and how to configure them, see xref:connectors::introduction/intro-connector-configuration-overview.adoc[Anypoint Connector Configuration].
. In the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

[[basic_auth]]
=== Basic Authentication

Enter the following information in the *General* tab of the *Global Element Properties* screen to configure Basic authentication:

[%header,cols="30s,70a"]
|===
|Field |Description
|Name | Configuration name.
|Vault URL | URL of the Veeva Vault instance required for the connection.
|Username | Username used to initialize the session.
|Password | Password used to authenticate the user.

> |Client Id | Client ID to send with each request header to the Veeva Vault instance for tracking purposes, in the format `\{organization}-{company/team}`. +

Reviewer: None of these descriptions should end with a period, because not one is a complete sentence.

> The client ID is prefixed with `mulesoft` and has a suffix with `client-vaultconnector` internally, for example, `mulesoft-abcpharma-clinicalprod-client-vaultconnector`.

Reviewer: I don't understand what we mean by "with `client-vaultconnector` internally."
```suggestion
The client ID has the prefix `mulesoft` and the suffix `client-vaultconnector`, for example, `mulesoft-abcpharma-clinicalprod-client-vaultconnector`.
```

image::basic-auth.png[Authentication fields completed in the *General* tab for Basic authentication]

Enter the following information in the *Advanced* tab of the *Global Element Properties* screen to configure the burst API limit for Basic authentication:

[%header,cols="30s,70a"]
|===
|Field |Description
|Response Timeout | Maximum amount of time, in milliseconds, that the connector waits for the Veeva Vault response. If the maximum amount of time is exceeded, a timeout exception is thrown. The default is 30,000 milliseconds (30 seconds).
|Burst API Limit Profile | Burst API limit profile configuration to act upon if the specified burst API limit of the Veeva Vault API reaches or exceeds the configured threshold. +

If *Burst API limit config* is enabled (default), then the connector evaluates the Veeva Vault API's response header (`X-VaultAPI-BurstLimitRemaining`) value against the specified *API Limit Threshold* value. If the remaining burst API limit reaches or exceeds the specified *API Limit Threshold* value, the connector waits for the specified amount of time to restart the count of *Burst Limit Remaining*. +

> The following example shows an application log with a warning message at `2019-06-06 10:34:45,096`, which waits for five minutes before rerunning as shown in the next message at `2019-06-06 10:39:45,098`:

Reviewer: ```suggestion
The following example shows an application log with a warning message at `2019-06-06 10:34:45,096`, halting the process for five minutes, as shown in the next message at `2019-06-06 10:39:45,098`:
```

INFO 2019-06-06 10:39:46,024 [[MuleRuntime].io.27: [veevavault-connector-demo].get-documents-operationFlow.BLOCKING @2dd2ffaf] [event: 99716950-8818-11e9-90e2-d61515f42856] org.mule.extension.veevavault.internal.service.HttpRequestService: Response received for sync http request :: 'https://[vault-domain]/api/v18.2/query?q=SELECT+name__v%2C+type__v%2C+subtype__v%2C+classification__v%2C+lifecycle__v%2C+status__v%2C+study__v%2C+product__v%2C+id+FROM+documents+WHERE+name__v%3D+%27F22611234--6764%27++LIMIT+1000+OFFSET+0' in 925 milliseconds.
----
|API Limit Threshold | Veeva Vault API limit threshold for *Burst API Limit Profile*. When the specified limit is reached or exceeded, the connector waits for the specified amount of time. The default is `200`.
|Wait Time | Amount of time that the connector waits if the *API Limit Threshold* is reached or exceeded. The default is `5`.
|Time Unit | Time unit for the wait time. The default is `MINUTES`.

> |Daily API Limit Profile | Daily API limit exceed profile configuration to use if the daily API limit of the Veeva Vault API reaches or exceeds the configured threshold. The daily limit is the total number of API calls within any 24-hour window on a daily rolling basis. +

Reviewer: Does it mean the same thing without the last phrase?
```suggestion
|Daily API Limit Profile | Daily API limit exceed profile configuration to use if the daily API limit of the Veeva Vault API reaches or exceeds the configured threshold. The daily limit is the total number of API calls within any 24-hour window. +
```
I'm not sure that I understand "on a daily rolling basis."

> If the *Daily API Limit Exceed Profile* is enabled (default), then the connector evaluates the Veeva Vault API's response header (`X-VaultAPI-DailyLimitRemaining`) for the specified value of the threshold, and throws an exception indicating that the daily API limit is exceeded. The exception is thrown for each job or each created request until the daily limit is reset or rolled over the next day. +

Reviewer: Is this para necessary? Doesn't it just repeat the previous?
```
